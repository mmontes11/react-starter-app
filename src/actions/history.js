import * as types from "constants/actionTypes/history";

export const setHistory = (squareIndex, symbol, step) => ({ type: types.SET_HISTORY, squareIndex, symbol, step });
