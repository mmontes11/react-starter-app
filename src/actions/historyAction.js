import * as types from "../constants/historyActionTypes";

export const setHistory = (squareIndex, symbol, step) => dispatch => {
  dispatch({ type: types.SET_HISTORY, squareIndex, symbol, step });
};
