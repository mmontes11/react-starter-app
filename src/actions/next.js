import * as types from "../constants/actionTypes/next";

export const nextPlayer = () => (dispatch, getState) => {
  dispatch({ type: types.NEXT_PLAYER, step: getState().step });
};
