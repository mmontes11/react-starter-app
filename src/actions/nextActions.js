import * as types from "../constants/nextActionTypes";

export const nextPlayer = () => (dispatch, getState) => {
  dispatch({ type: types.NEXT_PLAYER, step: getState().step });
};
