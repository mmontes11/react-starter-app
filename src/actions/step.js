import * as types from "../constants/actionTypes/step";

export const stepForward = () => ({ type: types.STEP_FORWARD });

export const jumpToStep = step => ({ type: types.JUMP_TO_STEP, step });
