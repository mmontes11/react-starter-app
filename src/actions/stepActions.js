import * as types from "../constants/stepActionTypes";

export const stepForward = () => ({ type: types.STEP_FORWARD });

export const jumpToStep = step => ({ type: types.JUMP_TO_STEP, step });
