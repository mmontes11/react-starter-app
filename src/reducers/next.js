import { NEXT_PLAYER } from "../constants/actionTypes/next";

const initialState = true;

const nextReducer = (state = initialState, action) => {
  switch (action.type) {
    case NEXT_PLAYER:
      return action.step % 2 === 0;
    default:
      return state;
  }
};

export default nextReducer;
