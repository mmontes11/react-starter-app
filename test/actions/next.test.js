import { nextPlayer } from "actions/next";

describe("actions/next", () => {
  describe("nextPlayer", () => {
    it("dispatches next player actions", () => {
      const thunk = nextPlayer();
      const dispatch = jest.fn();
      let step = 0;
      const getState = jest.fn().mockImplementation(() => {
        step += 1;
        return { step };
      });
      expect(thunk).toBeAThunk();
      thunk(dispatch, getState);
      thunk(dispatch, getState);
      expect(dispatch).toHaveBeenCalledTimes(2);
      expect(getState).toHaveBeenCalledTimes(2);
      expect(dispatch.mock.calls).toMatchSnapshot();
      expect(getState.mock.calls).toMatchSnapshot();
    });
  });
});
