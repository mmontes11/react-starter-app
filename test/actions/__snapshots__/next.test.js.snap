// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions/next nextPlayer dispatches next player actions 1`] = `
Array [
  Array [
    Object {
      "step": 1,
      "type": "NEXT_PLAYER",
    },
  ],
  Array [
    Object {
      "step": 2,
      "type": "NEXT_PLAYER",
    },
  ],
]
`;

exports[`actions/next nextPlayer dispatches next player actions 2`] = `
Array [
  Array [],
  Array [],
]
`;
